internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,357 @@
+		{"time_series_info.task", "perf"},
+		{"time_series_info.test", testname},
+		{"time_series_info.measurement", measurement},
+		{"last", true},
+		{"contexts", bson.D{{"$in", bson.A{"GoDriver perf task"}}}},
+	}
+
+	findOptions := options.FindOne().SetSort(bson.D{{"end", -1}})
+
+	findCtx, cancel := context.WithTimeout(context.Background(), 180*time.Second)
+	defer cancel()
+
+	var sr *StableRegion
+	err := coll.FindOne(findCtx, filter, findOptions).Decode(&sr)
+	if err != nil {
+		return nil, err
+	}
+	return sr, nil
+}
+
+// For a specific test and measurement
+func getEnergyStatsForOneBenchmark(rd RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
+	testname := rd.Info.TestName
+	var energyStats []*EnergyStats
+
+	for i := range rd.Rollups.Stats {
+		measurement := rd.Rollups.Stats[i].Name
+		patchVal := []float64{rd.Rollups.Stats[i].Val}
+
+		stableRegion, err := findLastStableRegion(testname, measurement, coll)
+		if err != nil {
+			log.Fatalf(
+				"Error finding last stable region for test %q, measurement %q: %v",
+				testname,
+				measurement,
+				err,
+			)
+		}
+
+		pChange := getPercentageChange(patchVal[0], stableRegion.Mean)
+		e, t, h := getEnergyStatistics(mat.NewDense(len(stableRegion.Values), 1, stableRegion.Values), mat.NewDense(1, 1, patchVal))
+		z := getZScore(patchVal[0], stableRegion.Mean, stableRegion.Std)
+
+		es := EnergyStats{
+			Benchmark:       testname,
+			Measurement:     measurement,
+			PatchVersion:    rd.Info.Version,
+			StableRegion:    *stableRegion,
+			PatchValues:     patchVal,
+			PercentChange:   pChange,
+			EnergyStatistic: e,
+			TestStatistic:   t,
+			HScore:          h,
+			ZScore:          z,
+		}
+		energyStats = append(energyStats, &es)
+	}
+
+	return energyStats, nil
+}
+
+func getEnergyStatsForAllBenchMarks(patchRawData []RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
+	var allEnergyStats []*EnergyStats
+	for _, rd := range patchRawData {
+		energyStats, err := getEnergyStatsForOneBenchmark(rd, coll)
+		if err != nil {
+			return nil, err
+		}
+		allEnergyStats = append(allEnergyStats, energyStats...)
+	}
+	return allEnergyStats, nil
+}
+
+func generatePRComment(energyStats []*EnergyStats, version string) string {
+	var comment strings.Builder
+	comment.WriteString("# 👋GoDriver Performance\n")
+	fmt.Fprintf(&comment, "The following benchmark tests for version %s had statistically significant changes (i.e., |z-score| > 1.96):\n", version)
+	comment.WriteString("| Benchmark | Measurement | H-Score | Z-Score | % Change | Stable Reg | Patch Value |\n| --- | --- | --- | --- | --- | --- | --- |\n")
+
+	var testCount int64
+	for _, es := range energyStats {
+		if math.Abs(es.ZScore) > 1.96 {
+			testCount += 1
+			fmt.Fprintf(&comment, "| %s | %s | %.4f | %.4f | %.4f | Avg: %.4f, Med: %.4f, Stdev: %.4f | %.4f |\n", es.Benchmark, es.Measurement, es.HScore, es.ZScore, es.PercentChange, es.StableRegion.Mean, es.StableRegion.Median, es.StableRegion.Std, es.PatchValues[0])
```

Comment:
You may give https://pkg.go.dev/text/tabwriter a shot, though I've never used it.